File: baekjoon/2884.js
const readline = require('readline')
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

let input = []

rl.on('line', function (line) {
  input.push(line.split(' '))
}).on('close', function () {
  let [H, M] = input[0]

  M = M - 45
  if (M < 0) {
    M = M + 60
    H = H - 1
    if (H === -1) H = 23
  }
  console.log(`${H} ${M}`)
  process.exit()
})

File: baekjoon/if문/2884.py
H, M = map(int,input().split())
calH = 0
calM = 0
if(M<45):
  calM = 60 - (45 - M)
  if(H == 0):
    calH = 23
  else:
    calH = H - 1
else:
  calM = M - 45
  calH = H
print(f"{calH} {calM}")